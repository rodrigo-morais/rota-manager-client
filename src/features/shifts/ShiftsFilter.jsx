import PropTypes from 'prop-types'
import React from 'react'

const ShiftsFilter = (props) => {
  const {jobTypes, selectedJobType, onChangeJobTypes, startTime, onChangeTimePeriod} = props

  const jobTypesOpetions = (types) =>
    types.map(type => <option key={type}>{type}</option>)

  return (
    <div>
      <h1>Shifts Filter</h1>
      <form>
        <label htmlFor='jobTypes'>Job Type: </label>
        <select id='jobTypes' value={selectedJobType} onChange={onChangeJobTypes}>
          {jobTypesOpetions(jobTypes)}
        </select>
        <label htmlFor='startTimes'>    Start Time: </label>
        <select id='startTimes' value={startTime} onChange={onChangeTimePeriod}>
          <option key='All'>All</option>
          <option key='AM'>AM</option>
          <option key='PM'>PM</option>
        </select>
      </form>
    </div>
  )
}

ShiftsFilter.propTypes = {
  jobTypes: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  selectedJobType: PropTypes.string.isRequired,
  onChangeJobTypes: PropTypes.func.isRequired,
  startTime: PropTypes.string.isRequired,
  onChangeTimePeriod: PropTypes.func.isRequired,
}

export default ShiftsFilter
