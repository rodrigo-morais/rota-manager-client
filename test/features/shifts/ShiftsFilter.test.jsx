import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import ShiftsFilter from '../../../src/features/shifts/ShiftsFilter'

const onChangeJobTypes = jest.fn()
const onChangeTimePeriod = jest.fn()

const jobTypes = ['waiter']

describe('ShiftsFilter', () => {
  let wraper

  beforeEach(() => {
    wraper = (
      <ShiftsFilter
        jobTypes={jobTypes}
        selectedJobType='All'
        onChangeJobTypes={onChangeJobTypes}
        startTime='AM'
        onChangeTimePeriod={onChangeTimePeriod}
      />
    )
  })

  describe('Rendering', () => {
    test('renders according to design', () => {
      const component = renderer.create(wraper)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('onChange callback', () => {
    beforeEach(() => {
      onChangeJobTypes.mockClear()
      onChangeTimePeriod.mockClear()
    })

    it('execute onChangeJobTypes callback when select a job type', () => {
      const component = mount(wraper)
      component.find('select').first().simulate('change', {target: { value: 'waiter' } })
      expect(onChangeJobTypes).toBeCalled()
    })

    it('execute onChangeTimePeriod callback when select a time start', () => {
      const component = mount(wraper)
      component.find('select').last().simulate('change', {target: { value: 'PM' } })
      expect(onChangeTimePeriod).toBeCalled()
    })
  })
})
