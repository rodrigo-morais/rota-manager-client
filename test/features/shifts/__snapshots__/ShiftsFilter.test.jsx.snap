// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShiftsFilter Rendering renders according to design 1`] = `
<div>
  <h1>
    Shifts Filter
  </h1>
  <form>
    <label
      htmlFor="jobTypes"
    >
      Job Type: 
    </label>
    <select
      id="jobTypes"
      onChange={[MockFunction]}
      value="All"
    >
      <option>
        waiter
      </option>
    </select>
    <label
      htmlFor="startTimes"
    >
          Start Time: 
    </label>
    <select
      id="startTimes"
      onChange={[MockFunction]}
      value="AM"
    >
      <option>
        All
      </option>
      <option>
        AM
      </option>
      <option>
        PM
      </option>
    </select>
  </form>
</div>
`;
